eng/Directory.Build.Common.targets

```diff
@@ -89,7 +89,7 @@
 
   <!-- Add Client SDK Tools -->
   <ItemGroup>
-    <PackageReference Condition="'$(IsClientLibrary)' == 'true'" Include="Microsoft.Azure.AutoRest.CSharp" PrivateAssets="All" />
+    <PackageReference Condition="'$(IncludeAutorestDependency)' == 'true'" Include="Microsoft.Azure.AutoRest.CSharp" PrivateAssets="All" />
```

Comment:
This is a temporary property that will be used until all libraries have migrated to TypeSpec and the new generator. I considered calculating this based on the presence of the `emitterPackageJsonPath` key in the tspLocation.yaml, but thought this would be too disruptive as this would impact all build targets, not just the generate target.